pkg/convert/tf.go

```diff
@@ -1141,18 +1142,20 @@ func convertTemplateWrapExpr(state *convertState,
 	return tokens
 }
 
+// convertTemplateJoinExpr writes "template joins" back out as they are.  These
+// are for expressions in the HCL template syntax like so:
+// "%{for foo in local.bars~}foo%{endfor~}
 func convertTemplateJoinExpr(state *convertState, inBlock bool, scopes *scopes,
 	fullyQualifiedPath string, expr *hclsyntax.TemplateJoinExpr,
 ) hclwrite.Tokens {
-	forExpr := expr.Tuple.(*hclsyntax.ForExpr)
-	if forExpr == nil {
+	forExpr, ok := expr.Tuple.(*hclsyntax.ForExpr)
+	if !ok {
```

Comment:
I'm still not convinced here -- couldn't a template also be e.g. an `IfExpr`? https://developer.hashicorp.com/terraform/language/expressions/strings#string-templates 

Or do we handle those elsewhere/am I missing the point sorry

Reply:
This is a different kind of expression (though I'm not sure which).  I added that case to the template_join test though